distribution/docker/Dockerfile

```diff
@@ -47,12 +47,13 @@ RUN ["/busybox/busybox", "--install", "/bin"]
 
 RUN addgroup -S -g 1000 druid \
  && adduser -S -u 1000 -D -H -h /opt/druid -s /bin/sh -g '' -G druid druid \
- && mkdir -p /opt/druid/var \
- && chown -R druid:druid /opt \
- && chmod 775 /opt/druid/var
+ && mkdir -p /opt
 
 COPY --chown=druid:druid --from=builder /opt /opt
 COPY distribution/docker/druid.sh /druid.sh
+RUN mkdir -p /opt/druid/var \
```

Comment:
nit, I would remove the `-p` to ensure `/opt/druid` exists and symlinks to a valid directory. When I did some testing, I realized that Docker for Mac gets confused when copying from the builder stage if `/opt/druid` is already created beforehand as a directory, causing it to fail to create the symlink.

Reply:
fixed

Reply:
Not sure this is the appropriate place for this question since this has been like it since the dockerfile was added afaict, but what is the reason for using the symlink for `/opt/druid`? Based on some comments around this issue/PR, and also looking at #11278, it seems to cause issues in some situations, so is it actually necessary for some reason or have some benefit?

Reply:
I checked the history of this file, the first commit used symlink there. I have no idea why symlink is used. 

I guess:
The name of raw directory contains a version variable whose value is retrieve by executing maven command against the source directory, while the value of this version variable could not be accessed in the following stage.